src/wrappers/koa/KoaWrapper.ts

```diff
@@ -0,0 +1,101 @@
+
+    ApplicationManager.setApplicationInfoProvider().update({
+        applicationClassName: ClassNames.KOA,
+        applicationDomainName: DomainNames.API,
+    });
+
+    const appInfo = ApplicationManager.getApplicationInfo();
+    ApplicationManager.getApplicationInfoProvider().update({
+        applicationId: WrapperUtils.getDefaultApplicationId(appInfo),
+    });
+
+    const config = opts.config || ConfigProvider.thundraConfig;
+    const {apiKey} = config;
+    const reporter = opts.reporter || new Reporter(apiKey);
+    const pluginContext = opts.pluginContext || WrapperUtils.createPluginContext(apiKey, KoaExecutor);
+    const plugins = opts.plugins || WrapperUtils.createPlugins(config, pluginContext);
+
+    if (!opts.disableAsyncContextManager) {
+        WrapperUtils.initAsyncContextManager();
+    }
+
+    ThundraLogger.debug('<KoaWrapper> Creating Thundra middleware ...');
+
+    return async (ctx: any, next: any) => ExecutionContextManager.runWithContext(
+        WrapperUtils.createExecContext,
+        async function () {
+            ThundraLogger.debug('<KoaWrapper> Running with execution context');
+            const context: ExecutionContext = this;
+            ctx.thundra = {
+                executionContext: context,
+                setError(err: any) {
+                    context.error = Utils.buildError(err);
+                },
+                report() {
+                    ExecutionContextManager.set(context);
+                    ThundraLogger.debug('<KoaWrapper> Reporting request');
+                    WrapperUtils.afterRequest(ctx.request, ctx.response, plugins, reporter);
+                },
+            };
+            try {
+                await WrapperUtils.beforeRequest(ctx.request, ctx.response, plugins);
+                ctx.res.once('finish', () => {
+                    ctx.request._matchedRoute = ctx._matchedRoute;
+                    ExecutionContextManager.set(context);
+                    WrapperUtils.afterRequest(ctx.request, ctx.response, plugins, reporter);
+                });
+            } catch (err) {
+                ThundraLogger.error('<KoaWrapper> Error occurred in KoaWrapper:', err);
+            } finally {
+                ThundraLogger.debug('<KoaWrapper> Calling next middleware');
+                await next();
+            }
+        },
+    );
+}
+
+function wrapUse(originalUse: Function) {
+    ThundraLogger.debug('<KoaWrapper> Wrapping "app.use" ...');
+    return function useWrapper() {
+        const result = originalUse.apply(this, arguments);
+        if (this._thundra) {
+            return result;
+        }
+        this._thundra = true;
+        this.on('error', async (err: any, ctx: any) => {
+            if (ctx.thundra) {
+                ctx.thundra.setError(err);
+            }
+        });
+        return result;
+    };
+}
+
+export function init() {
+    ThundraLogger.debug('<KoaWrapper> Initializing ...');
```

Comment:
I think, we should have a config (`thundra.agent.trace.integrations.koa.disable`) to disable Koa wrapper explicitly in case of Koa wrapper issue